src/Role/ReadModel/Permissions/AppConfigUserPermissionsReadRepository.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Role\ReadModel\Permissions;
+
+use CultuurNet\UDB3\Role\ValueObjects\Permission;
+
+final class AppConfigUserPermissionsReadRepository implements UserPermissionsReadRepositoryInterface
+{
+    /**
+     * @var array<string, Permission[]>
+     */
+    private array $clientIdToPermissions;
+
+    private UserPermissionsReadRepositoryInterface $userPermissionsReadRepository;
+
+
+    public function __construct(UserPermissionsReadRepositoryInterface $userPermissionsReadRepository, array $clientIdToPermissionsConfig)
+    {
+        $this->userPermissionsReadRepository = $userPermissionsReadRepository;
+        $this->clientIdToPermissions = $clientIdToPermissionsConfig;
+    }
+
+
+    public function getPermissions(string $userId): array
+    {
+        return $this->clientIdToPermissions[$userId] ??
+            $this->userPermissionsReadRepository->getPermissions($userId);
+    }
+
+    public function hasPermission(string $userId, Permission $permission): bool
+    {
+        return in_array($permission, $this->getPermissions($userId));
```

Comment:
Should this also not delegate to `userPermissionsReadRepository` to check if it has the correct permission?

Reply:
Ah, not needed because it uses `getPermissions`